scripts/test-versions.sh

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env bash
+set -ueo pipefail
+
+main_cfg_dir=$1
+command=$2
+
+bucket="bucket-123"
+test_files_dir="${main_cfg_dir}/testfiles"
+stage1_dst_dir="${main_cfg_dir}/stage1"
```

Comment:
just to clarify - I think stage 1 is used for making sure that uploads work across different versions, and stage 2 is used to check that downloads work across different versions. is that right?